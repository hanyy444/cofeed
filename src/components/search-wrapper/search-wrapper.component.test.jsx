import { MemoryRouter } from "react-router-dom";
import * as router from 'react-router'
import { render, screen } from "@testing-library/react";
import SearchWrapper from "./search-wrapper.component";
import User from "../display/user/user.component";
import Spinner from "../display/spinner/spinner.component";

// beforeEach(()=>{
//     vi
//         .spyOn(router, 'useNavigate')
//         .mockImplementationOnce(()=>vi.fn())
// })

describe('Test search-wrapper component', () => {     

    afterEach(() => {
        vi.clearAllMocks()
    })

    it('Should render without crashing', () => {
        expect.assertions(2)
        render(
            <MemoryRouter>
                <SearchWrapper />
            </MemoryRouter>
        )
        const searchWrapper = screen.getByTestId('search-wrapper')
        expect(searchWrapper).toBeInTheDocument()
        expect(searchWrapper).toMatchSnapshot()
    })

    it('Should render 1 result', () => {
        expect.assertions(2)
        render(
            <MemoryRouter>
                <SearchWrapper data={[{ _id: 'fake-id', firstName: 'test' }]} loading="success"/>
            </MemoryRouter>
        )
        const searchWrapper = screen.getByTestId('search-wrapper')
        expect(searchWrapper).toContainElement(screen.getByTestId('user'))
        expect(searchWrapper).toHaveTextContent('test')
    })

    it('Should render 0 results', () => {
        expect.assertions(1)
        render(
            <MemoryRouter>
                <SearchWrapper data={[]} loading="success" count={0}/>
            </MemoryRouter>
        )
        const searchWrapper = screen.getByTestId('search-wrapper')
        expect(searchWrapper).toHaveTextContent('0 results found')
    })

    it('Should return spinner on loading', () => {
        expect.assertions(1)
        render(
            <MemoryRouter>
                <SearchWrapper count={0} loading='pending'/>
            </MemoryRouter>
        )
        const searchWrapper = screen.getByTestId('search-wrapper')
        expect(searchWrapper).toContainElement(screen.getByTestId('spinner'))
    })
 })