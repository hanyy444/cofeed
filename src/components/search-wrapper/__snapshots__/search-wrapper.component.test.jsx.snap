// Vitest Snapshot v1

exports[`Test search-wrapper component > Should render without crashing 1`] = `
<div
  class="search-wrapper"
  data-testid="search-wrapper"
  id="search-wrapper"
>
  <div
    class="spinner"
    data-testid="spinner"
    id="spinner"
  />
</div>
`;
